operator-common/src/main/java/io/strimzi/operator/common/model/PemAuthIdentity.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.model;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.operator.common.Util;
+
+import java.io.ByteArrayInputStream;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+
+/**
+ * Represents the identity used during TLS client authentication.
+ * This consists of an X509 end-entity certificate, corresponding private key, and a (possibly empty) chain of X509 intermediate CA certificates, all in PEM format.
+ */
+public class PemAuthIdentity {
+
+    private final byte[] privateKeyAsPemBytes;
+    private final byte[] certificateChainAsPemBytes;
+    private final String secretCertName;
+    private String secretName;
+    private String secretNamespace;
+
+    /**
+     * @param secret Kubernetes Secret containing the Cluster Operator public and private key
+     * @param secretCertName Key in the Kubernetes Secret that is associated with the requested identity
+     */
+    private PemAuthIdentity(Secret secret, String secretCertName) {
+        this.secretCertName = secretCertName;
+        Optional.ofNullable(secret)
+                .map(Secret::getMetadata)
+                .ifPresent(objectMeta -> {
+                    secretName = objectMeta.getName();
+                    secretNamespace = objectMeta.getNamespace();
+                });
```

Comment:
Kube resources should always have their metadata. So I think we should not do this just if present if we really need this.

Reply:
I've removed this